typescript/sdk/src/gcp/storage.ts

```diff
@@ -18,66 +16,91 @@ export interface StorageConfig {
 }
 
 export class GcpStorageWrapper {
-  private readonly client: Storage;
   private readonly bucket: string;
   private cache: Record<string, StorageReceipt<any>> | undefined;
+  private readonly baseUrl: string;
 
   static fromBucketUrl(bucketUrl: string): GcpStorageWrapper {
     const match = bucketUrl.match(GCS_BUCKET_REGEX);
     if (!match) throw new Error('Could not parse bucket url');
     return new GcpStorageWrapper({
-      bucket: match[1],
+      // Handle both http and gs:// formats
+      bucket: match[1] || match[2],
       caching: true,
     });
   }
 
   constructor(readonly config: StorageConfig) {
-    this.client = new Storage({
-      projectId: config.projectId,
-      keyFilename: config.keyFilename,
-    });
     this.bucket = config.bucket;
+    this.baseUrl = `https://storage.googleapis.com/storage/v1/b/${this.bucket}/o`;
     if (config.caching) {
       this.cache = {};
     }
   }
 
-  formatKey(key: string): string {
+  private formatKey(key: string): string {
     return this.config.folder ? `${this.config.folder}/${key}` : key;
   }
 
-  async getObject<T>(key: string): Promise<StorageReceipt<T> | undefined> {
-    const Key = this.formatKey(key);
-    if (this.cache?.[Key]) {
-      return this.cache![Key];
+  private getCachedObject<T>(key: string): StorageReceipt<T> | undefined {
+    return this.cache?.[key];
+  }
+
+  private setCachedObject<T>(key: string, value: StorageReceipt<T>): void {
+    if (this.cache) {
+      this.cache[key] = value;
     }
+  }
 
-    try {
-      const bucket = this.client.bucket(this.bucket);
-      const file = bucket.file(Key);
-      const [exists] = await file.exists();
+  private async fetchMetadata(key: string): Promise<any> {
+    const url = `${this.baseUrl}/${encodeURIComponent(key)}`;
+    const response = await fetch(url);
 
-      if (!exists) {
-        return undefined;
-      }
+    if (response.status === 404) return undefined;
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to fetch object metadata: ${response.statusText}`,
+      );
```

Comment:
I think it would be good to log the error response too because it might include useful info on how to solve the error if it is something simple